src/Agent.Sdk/Util/IOUtil.cs

```diff
@@ -14,26 +15,18 @@ public static class IOUtil
     {
         public static string ExeExtension
         {
-            get
-            {
-#if OS_WINDOWS
-                return ".exe";
-#else
-                return string.Empty;
-#endif
-            }
+            get =>
+                PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows
+                ? ".exe"
+                : string.Empty;
         }
 
         public static StringComparison FilePathStringComparison
         {
-            get
-            {
-#if OS_LINUX
-                return StringComparison.Ordinal;
-#else
-                return StringComparison.OrdinalIgnoreCase;
-#endif
-            }
+            get =>
+                PlatformUtil.RunningOnOS == PlatformUtil.OS.Linux
```

Comment:
I think this is a bug. Both Linux and OSx are case-sensitive

Reply:
this might be your Win-X86 bug

Reply:
macOS is case-preserving, not case-sensitive. This matches the logic from before, which read:

```csharp
            get
            {
#if OS_LINUX
                return StringComparison.Ordinal;
#else
                return StringComparison.OrdinalIgnoreCase;
#endif
            }
```

(The Win-x86 thing is because our agents are offline, not a logic bug.)